ros2doctor/ros2doctor/api/package.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import xml.etree.ElementTree as ET
+
+from ament_index_python import get_package_prefix
+from ament_index_python.packages import PackageNotFoundError
+from ros2doctor.api import DoctorCheck
+from ros2doctor.api import DoctorReport
+from ros2doctor.api import Report
+from ros2doctor.api import Result
+from ros2doctor.api.format import doctor_warn
+
+import rosdistro
+
+
+def _get_ros_packages_info(*, check=True) -> dict:
+    """
+    Return all current distro's packages info using rosdistro API.
+
+    :return: a dictionary contains package name, release, source and status
+    """
+    distro_name = os.environ.get('ROS_DISTRO')
+    if not distro_name:
+        if check:  # suppress warning when printing report
+            doctor_warn('ROS_DISTRO is not set.')
+        return
+    distro_name = distro_name.lower()
+    url = rosdistro.get_index_url()
+    if not url:
+        if check:  # suppress warning when printing report
+            doctor_warn('Unable to access ROSDISTRO_INDEX_URL or DEFAULT_INDEX_URL.')
+        return
+    i = rosdistro.get_index(url)
+    distro_data = rosdistro.get_distribution(i, distro_name).get_data()
+    return distro_data.get('repositories')
+
+
+def _get_local_package_version(package_name, *, check=True) -> str:
+    """
+    Return all locally installed packages info.
+
+    :param: str of package name
+    :return: a string contains local package version
+    """
+    try:
+        prefix = get_package_prefix(package_name)
+    except PackageNotFoundError:
+        if check:  # suppress warning when printing report
+            doctor_warn('Unable to find prefix for `%s` using index system.' % package_name)
+        return
+    xml_file = os.path.join(prefix, 'share', package_name, 'package.xml')
+    try:
+        root = ET.parse(xml_file).getroot()
```

Comment:
Instead of parsing the manifest file yourself please use the API provided by the Python package `catkin_pkg` (rosdep key `python3-catkin-pkg-modules`) instead.